src/commands/TeamCommand.ts

```diff
@@ -65,15 +67,36 @@ export default class TeamCommand implements Command {
 
         const method = interaction.options.getString("method");
         if (method === "random") {
-          this.randomTeams.randomizeTeams();
-          const response = this.randomTeams.createEmbedMessage();
+          game.shuffleTeams("random");
+          const response = createTeamGenerateEmbed(game);
           await interaction.reply(response);
         } else {
           await interaction.reply({
             content: "Invalid generation method!",
             ephemeral: true,
           });
         }
+
+        const redTeam = game.getPlayersOfTeam("RED");
+        for (let i = 0; i < redTeam.length; i++) {
+          const player = redTeam[i];
+          const discordUser = await interaction.guild?.members.fetch(
+            player.discordSnowflake
+          );
+          await discordUser?.roles.remove(config.roles.blueTeamRole);
+          discordUser?.roles.add(config.roles.redTeamRole);
+        }
+
+        const blueTeam = game.getPlayersOfTeam("BLUE");
+        for (let i = 0; i < blueTeam.length; i++) {
+          const player = blueTeam[i];
+          const discordUser = await interaction.guild?.members.fetch(
+            player.discordSnowflake
+          );
+          await discordUser?.roles.remove(config.roles.redTeamRole);
+          discordUser?.roles.add(config.roles.blueTeamRole);
+        }
```

Comment:
This needs to be delayed until the organiser clicks the confirm button  because if they randomise the teams several teams discord will probs rate limit bot and that'll mess everything up if it happens even once

Reply:
might need to add a delay in general during the loop or at least between teams idk